app/view_models/auctions_index_view_model.rb

```diff
@@ -1,29 +1,30 @@
-class AuctionsIndexViewModel < Struct.new(:current_user, :auctions_query)
-  def auctions
-    @auctions ||= auctions_query.map { |auction| AuctionViewModel.new(current_user, auction) }
+class AuctionsIndexViewModel
+  attr_reader :auctions, :current_user
+
+  def initialize(auctions:, current_user:)
+    @auctions = auctions.to_a
```

Comment:
Oh that is right. I would recommend though that you perhaps move the `to_a` into the `AuctionQuery#public_index` method to make it explicitly run there, because we aren't really designing the AuctionQuery objects to be chained, are we? And this delayed execution can lead to confusing errors being thrown from views or such.